internal/client.go

```diff
@@ -174,3 +174,32 @@ func (c *Client) Options(path string) (classes map[string]bool, methods map[stri
 	methods = parseCommaSeparatedSet(resp.Header["Allow"], true)
 	return classes, methods, nil
 }
+
+type SyncLevel string
+
+const (
+	SyncLevel1        SyncLevel = "1"
+	SyncLevelInfinite SyncLevel = "infinite"
+)
+
+// SyncCollection perform a `sync-collection` REPORT operation on a resource
+func (c *Client) SyncCollection(path, syncToken string, level Depth, limit *Limit, prop *Prop) (*Multistatus, error) {
+	q := &SyncCollectionQuery{
```

Comment:
No need to take the address here with `&`, because that results in two indirections with the additional `&` in the `NewXMLRequest` call below.